internal/service/advancedclustertpf/model_ClusterDescription20240805.go

```diff
@@ -0,0 +1,276 @@
+		TerminationProtectionEnabled:              types.BoolPointerValue(input.TerminationProtectionEnabled),
+		VersionReleaseSystem:                      types.StringPointerValue(input.VersionReleaseSystem),
+		Timeouts:                                  timeout,
+	}
+}
+
+func NewBiConnectorObjType(ctx context.Context, input *admin.BiConnector, diags *diag.Diagnostics) types.Object {
+	if input == nil {
+		return types.ObjectNull(BiConnectorObjType.AttrTypes)
+	}
+	tfModel := TFBiConnectorModel{
+		Enabled:        types.BoolPointerValue(input.Enabled),
+		ReadPreference: types.StringPointerValue(input.ReadPreference),
+	}
+	objType, diagsLocal := types.ObjectValueFrom(ctx, BiConnectorObjType.AttrTypes, tfModel)
+	diags.Append(diagsLocal...)
+	return objType
+}
+
+func NewConnectionStringsObjType(ctx context.Context, input *admin.ClusterConnectionStrings, diags *diag.Diagnostics) types.Object {
+	if input == nil {
+		return types.ObjectNull(ConnectionStringsObjType.AttrTypes)
+	}
+	privateEndpoint := NewPrivateEndpointObjType(ctx, input.PrivateEndpoint, diags)
+	tfModel := TFConnectionStringsModel{
+		AwsPrivateLink:    conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLink),
+		AwsPrivateLinkSrv: conversion.ToTFMapOfString(ctx, diags, input.AwsPrivateLinkSrv),
+		Private:           types.StringPointerValue(input.Private),
+		PrivateEndpoint:   privateEndpoint,
+		PrivateSrv:        types.StringPointerValue(input.PrivateSrv),
+		Standard:          types.StringPointerValue(input.Standard),
+		StandardSrv:       types.StringPointerValue(input.StandardSrv),
+	}
+	objType, diagsLocal := types.ObjectValueFrom(ctx, ConnectionStringsObjType.AttrTypes, tfModel)
+	diags.Append(diagsLocal...)
+	return objType
+}
+
+func NewLabelsObjType(ctx context.Context, input *[]admin.ComponentLabel, diags *diag.Diagnostics) types.Set {
+	if input == nil {
+		return types.SetNull(LabelsObjType)
+	}
+	tfModels := make([]TFLabelsModel, len(*input))
+	for i, item := range *input {
+		tfModels[i] = TFLabelsModel{
+			Key:   types.StringPointerValue(item.Key),
+			Value: types.StringPointerValue(item.Value),
+		}
+	}
+	setType, diagsLocal := types.SetValueFrom(ctx, LabelsObjType, tfModels)
+	diags.Append(diagsLocal...)
+	return setType
+}
+
+func NewMongoDBEmployeeAccessGrantObjType(ctx context.Context, input *admin.EmployeeAccessGrant, diags *diag.Diagnostics) types.Object {
+	if input == nil {
+		return types.ObjectNull(MongoDbEmployeeAccessGrantObjType.AttrTypes)
+	}
+	tfModel := TFMongoDbEmployeeAccessGrantModel{
+		ExpirationTime: types.StringValue(conversion.TimeToString(input.ExpirationTime)),
+		GrantType:      types.StringValue(input.GrantType),
+	}
+	objType, diagsLocal := types.ObjectValueFrom(ctx, MongoDbEmployeeAccessGrantObjType.AttrTypes, tfModel)
+	diags.Append(diagsLocal...)
+	return objType
+}
+
+func NewReplicationSpecsObjType(ctx context.Context, input *[]admin.ReplicationSpec20240805, diags *diag.Diagnostics) types.List {
```

Comment:
what are the pros and cons of passing input as pointer or directly the slice? same for all functions

Reply:
Pros:
1. That is the only approach that can separate between a non-existing (`nil`) that was not part of a response
2. Performance

Reply:
not sure Performance is an issue here, ok 1 about missing.    Just to confirm, is a ListNull different from an empty list? do we want to make a difference between input == nil and input == &[]admin.ReplicationSpec20240805{} ?

Reply:
Yes, ListNull != empty list.
Yes, we want to make a difference. 
A nil object will not be dumped in the json

Reply:
nice, thanks for the explanation

Reply:
although similarly to maps, Go can have a nil or a 0-element slice, wouldn't that be enough?

![slice](https://github.com/user-attachments/assets/a2507bdf-f46f-4dd8-9069-bd7883f734fb)

Reply:
i suppose pointers are used in SDK and here we're just bringing it here, it's fine in that case

Reply:
Yes, it comes from the SDK